File: algorithms/algorithms-cplusplus/leetcode/Question_0274_H-Index.cc
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
 public:
  int hIndex(vector<int>& citations) {
    sort(citations.begin(), citations.end());
    for (int i = citations.size() - 1; i >= 0; i--) {
      if (citations.size() - 1 - i >= citations[i]) {
        return citations.size() - 1 - i;
      }
    }
    return citations.size();
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0274_H_Index.java
package com.lanceolata.coding.algorithms.java.leetcode;

import java.util.Arrays;

public class Question_0274_H_Index {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);
        for (int i = citations.length - 1; i >= 0; i--) {
            if (citations.length - 1 - i >= citations[i]) {
                return citations.length - 1 - i;
            }
        }
        return citations.length;
    }
}
